src/screens/FeedScreen/RenderList.js

```diff
@@ -182,55 +242,43 @@ const RenderListFeed = (props) => {
             isSelf={item.anonimity ? false : selfUserId === item.actor.id}
           />
         </View>
-        {hasComment ? (
-          <View testID="previewComment" style={styles.contentReaction(getHeightReaction())}>
-            <BlurredLayer
-              layerOnly
-              blurType="light"
-              withToast={true}
-              isVisible={item?.isBlurredPost && item?.anonimity}>
-              <React.Fragment>
-                <PreviewComment
-                  user={item.latest_reactions.comment[0].user}
-                  comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
-                  image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
-                  time={item.latest_reactions.comment[0].created_at}
-                  totalComment={getTotalReaction(item) - 1}
-                  item={item.latest_reactions.comment[0]}
-                  onPress={onPress}
-                />
-                <Gap height={8} />
-              </React.Fragment>
-            </BlurredLayer>
-          </View>
-        ) : (
-          <TouchableOpacity
-            testID="writeComment"
-            onPress={() => onPressComment(isHaveSeeMore)}
-            style={styles.contentReaction(getHeightReaction())}>
-            <WriteComment
-              postId={''}
-              username={handleUserName(item)}
-              value={''}
-              onChangeText={() => {}}
-              onPress={() => {}}
-              loadingPost={false}
-              isViewOnly={true}
-              withAnonymityLabel={false}
-            />
-          </TouchableOpacity>
-        )}
+        <BlurredLayer
+          layerOnly
+          blurType="light"
+          withToast={true}
+          isVisible={isBlurred}
+          containerStyle={{
+            height: commentHeight()
+          }}>
+          {hasComment ? (
+            <View
+              testID="previewComment"
+              style={styles.contentReaction(getHeightReaction(), isBlurred)}>
+              <PreviewComment
+                user={item.latest_reactions.comment[0].user}
+                comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
+                image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
+                time={item.latest_reactions.comment[0].created_at}
+                totalComment={getTotalReaction(item) - 1}
+                item={item.latest_reactions.comment[0]}
+                onPress={onPress}
+              />
+            </View>
+          ) : (
+            renderWriteComment()
+          )}
+        </BlurredLayer>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for rendering blurred content is quite complex. Consider simplifying it or breaking it down into smaller, more manageable functions. This will make the code easier to understand and maintain. Also, ensure that all the functions like `commentHeight()`, `onPress` are properly handling potential null or undefined values to avoid runtime errors.



<!-- This is an auto-generated comment by OSS CodeRabbit -->